subprojects/docs/src/snippets/java/toolchain-management/groovy/settings.gradle

```diff
@@ -0,0 +1,65 @@
+import java.util.Optional
+import javax.inject.Inject
+
+apply plugin: AzulPlugin
+apply plugin: AdoptiumPlugin
+
+// tag::toolchain-management[]
+toolchainManagement {
+    jdks { // <1>
+        add("com_azul_zulu") // <2>
+            {
+                credentials {
+                    username "user"
+                    password "password"
+                }
+                authentication {
+                    digest(BasicAuthentication)
+                }
+            } // <3>
+        add("org_gradle_adopt_open_jdk") // <4>
```

Comment:
I know this is an implementation detail, but the use of underscores is quite ugly. Wouldn't it be possible to automatically convert dots to underscores?

Reply:
We are planning on generating accessors for these names (right now it doesn't work at settings level, but we'll fix that too). That is what produced the initial constraints, but yes, we will think about improvements.